lib/ronin/web/browser/agent.rb

```diff
@@ -79,10 +83,17 @@ def initialize(visible:  false,
                     raise(ArgumentError,"invalid proxy value (#{proxy.inspect}), must be either a Hash, URI::HTTP, String, or nil")
                   end
 
+          url = case url
+                when nil, String then url
+                else
+                  raise(ArgumentError,"invalid url value (#{url.inspect}), must be either String, or nil")
+                end
```

Comment:
Is this code really necessary? We could let `go_to()` just try to handle the `url` value if it's non-`nil`.

If it is needed, you could move the `case` statement down below around the `go_to`.

```ruby
case url
when String then go_to(url)
when nil then # no-op
else
  ...
end
```

Reply:
Have refactored with your revised case statement rather than removed - the error messaging does seem much nicer/clearer from a user perspective when raising the ArgumentError, rather than allowing Ferrum to handle it. Just let me know if you disagree and can remove.